web/src/toga_web/widgets/timeinput.py

```diff
@@ -0,0 +1,57 @@
+import datetime
+
+from toga_web.libs import create_proxy
+
+from .base import Widget
+
+
+def py_time(native_time):
+    return datetime.time.fromisoformat(native_time)
+
+
+def native_time(py_time):
+    return py_time.strftime("%H:%M")
+
+
+class TimeInput(Widget):
+    def create(self):
+        self._return_listener = None
+        self.native = self._create_native_widget("sl-input")
+        self.native.setAttribute("type", "time")
+
+        self.set_value(datetime.datetime.now().time().strftime("%H:%M"))
+        self.native.addEventListener("sl-change", create_proxy(self.on_change))
+
+    def on_change(self, event):
+        self.interface.on_change(None)
+
+    def get_value(self):
+        return py_time(self.native.value)
+
+    def set_value(self, value):
+        if value is None:
+            self.native.value = ""
+        self.native.value = self._format_time(value)
+
+    def set_min_time(self, value):
+        self.native.min = self._format_time(value)
+
+    def set_max_time(self, value):
+        self.native.max = self._format_time(value)
```

Comment:
These min/max values don't appear to be applied - I can set times before the min/after the max in the example app.

Reply:
I'd love to know the way you'd like this handled, the way I'm thinking is to essentially set the value for that time to the minimum or maximum (which ever is relevant).

The only thing here is that browsers don't enforce a min/max on the time input and I couldn't find the max/min on the shoelace components.

I'm certainly open to anything here but I think it might be a bit hidden if we set the `value` here to the `max` or `min` time.

Reply:
FWIW, Android has a similar limitation. The approach we've taken there is to catch the "on change" event, and when it occurs, explicitly use the API to set the value of the widget to the value that was just set. This might seems like a no-op, but it has the effect of clipping the time to the min/max bounds.